File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    /*
         1 ->2 ->3 ->4 ->5
    */
    ListNode* reverseList(ListNode* head) {
        vector<ListNode*> v;
        ListNode* temp = head;
        while(temp){
            v.push_back(temp);
            temp = temp->next;
        }
        int n = v.size();
        for(int i = n-1; i >= 0; i--){
            if(i == 0){
                v[i]->next = NULL;
                break;
            }
            v[i]->next = v[i-1];
        }
        if(n == 0){
            return NULL;
        }
        return v[n-1];
    }
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev, forward, curr = None, None, head
        
        #forward = None
        while curr != None:
            forward = curr.next
            curr.next = prev
            prev = curr
            curr = forward

        return prev
            
        